test/e2e/broker_event_transformation_test.go

```diff
@@ -0,0 +1,157 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+/*
+TestEventTransformationForTrigger tests the following scenario:
+
+                         5                 4
+                   ------------- ----------------------
+                   |           | |                    |
+             1     v	 2     | v        3           |
+EventSource ---> Broker ---> Trigger1 -------> Service(Transformation)
+                   |
+                   | 6                   7
+                   |-------> Trigger2 -------> Service(Logger)
+
+Note: the number denotes the sequence of the event that flows in this test case.
+*/
+
+func TestEventTransformationForTrigger(t *testing.T) {
+	const (
+		brokerName  = "e2e-eventtransformation-broker"
+		rbacObjName = "eventing-broker-filter"
+
+		any          = v1alpha1.TriggerAnyFilter
+		eventType1   = "type1"
+		eventType2   = "type2"
+		eventSource1 = "source1"
+		eventSource2 = "source2"
+		eventBody    = "e2e-eventtransformation-body"
+
+		triggerName1 = "trigger1"
+		triggerName2 = "trigger2"
+
+		transformationPodName = "trans-pod"
+		loggerPodName         = "logger-pod"
+	)
+
+	clients, ns, provisioner, cleaner := Setup(t, true, t.Logf)
+	defer TearDown(clients, ns, cleaner, t.Logf)
+
+	// creates ServiceAccount and ClusterRoleBinding with default cluster-admin role
+	err := CreateServiceAccountAndBinding(clients, rbacObjName, ns, t.Logf, cleaner)
+	if err != nil {
+		t.Fatalf("Failed to create the ServiceAccount and ServiceAccountRoleBinding: %v", err)
+	}
+
+	// create a new broker
+	broker := test.Broker(brokerName, ns, test.ClusterChannelProvisioner(provisioner))
+	t.Logf("provisioner name is: %s", broker.Spec.ChannelTemplate.Provisioner.Name)
+	err = WithBrokerReady(clients, broker, t.Logf, cleaner)
+	if err != nil {
```

Comment:
some of this is duplicated from the `broker_trigger_test` ... not sure worth some sharing of code

Reply:
In the current test cases, we have a lot of duplicate code in the topology setup process, not just here. We are considering to develop a framework to create the topology from some meta data. But for now, I would prefer not to spend much time on refactoring it.